lib/model/message.dart

```diff
@@ -409,22 +409,59 @@ class MessageStoreImpl extends HasChannelStore with MessageStore, _OutboxMessage
     // from the event queue, so there's inherently a race.
     //
     // If the fetched message reflects changes we haven't yet heard from the
-    // event queue, then it doesn't much matter which version we use: we'll
-    // soon get the corresponding events and apply the changes anyway.
-    // But if it lacks changes we've already heard from the event queue, then
-    // we won't hear those events again; the only way to wind up with an
+    // event queue, then normally (see [1]) it doesn't much matter which version
+    // we use: we'll soon get the corresponding events and apply the changes
+    // anyway. But if it lacks changes we've already heard from the event queue,
+    // then we won't hear those events again; the only way to wind up with an
     // updated message is to use the version we have, that already reflects
-    // those events' changes.  So we always stick with the version we have.
+    // those events' changes.  So we always stick with the version we have. [1]
+    //
+    // [1] With one exception: if the version we have was in an unsubscribed
+    //     channel when we got it or sometime since, we take the fetched version
```

Comment:
I like this formulation of which messages go in `_maybeStaleChannelMessages`. It'd be good to put it in that field's doc.